public/app/features/explore/utils/query.ts

```diff
@@ -0,0 +1,81 @@
+import { DataQuery, AbstractQuery, AbstractLabelOperator, AbstractLabelMatcher } from '@grafana/data';
+import { Token } from 'prismjs';
+
+const FromPromLikeMap: Record<string, AbstractLabelOperator> = {
+  '=': AbstractLabelOperator.Equal,
+  '!=': AbstractLabelOperator.NotEqual,
+  '=~': AbstractLabelOperator.EqualRegEx,
+  '!~': AbstractLabelOperator.NotEqualRegEx,
+};
+const ToPromLikeMap: Record<AbstractLabelOperator, string> = {
+  [AbstractLabelOperator.Equal]: '=',
+  [AbstractLabelOperator.NotEqual]: '!=',
+  [AbstractLabelOperator.EqualRegEx]: '=~',
+  [AbstractLabelOperator.NotEqualRegEx]: '!~',
+};
+
+export function toPromLikeQuery(labelBasedQuery: AbstractQuery): PromLikeQuery {
+  const expr = labelBasedQuery.labelMatchers
+    .map((selector: AbstractLabelMatcher) => {
+      const operator = ToPromLikeMap[selector.operator];
+      if (operator) {
+        return `${selector.name}${operator}"${selector.value}"`;
+      } else {
+        return '';
+      }
+    })
+    .filter((e: string) => e !== '')
+    .join(', ');
+
+  return {
+    refId: labelBasedQuery.refId,
+    expr: expr ? `{${expr}}` : '',
+    range: true,
+  };
+}
+
+export interface PromLikeQuery extends DataQuery {
+  expr: string;
+  range: boolean;
+}
+
+export function extractLabelMatchers(tokens: Array<string | Token>): AbstractLabelMatcher[] {
```

Comment:
Also these seems to be used only in prometheus and loki, so I would move it outside of explore utils.

Reply:
It's used in Prometheus and Loki. I've moved it with toPromLikeQuery function to language_utils.